docs/docs/remark-ast.md

```diff
@@ -0,0 +1,270 @@
+
+The plugin is now initially set so we can add it as a sub-plugin inside `gatsby-transformer-remark`
+
+```js:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: "gatsby-source-filesystem",
+      options: {
+        name: `data`,
+        path: `${__dirname}/src/data/`,
+      },
+    },
+    {
+      resolve: `gatsby-transformer-remark`,
+      options: {
+        plugins: [`gatsby-remark-purple-headers`],
+      },
+    },
+  ],
+}
+```
+
+If we wanted to add some options, we could switch to the object syntax:
+
+```js
+{
+  resolve: `gatsby-remark-purple-headers`,
+  options: {
+    // Options here
+  }
+}
+```
+
+## Find and Modify Markdown Nodes
+
+When modifying nodes, you'll want to walk the tree and then implement new functionality on specific nodes.
+
+A node module to help with is [unist-util-visit](https://github.com/syntax-tree/unist-util-visit), a walker for `unist` nodes. For reference, Unist which stands for United Syntax Tree is a standard for markdown syntax trees and parsers that include well known parsers in the Gatsby world like Remark and MDX.
+
+As an example from `unist-util-visit`'s readme, it allows for an easy interface to visit particular nodes based on a particular type:
+
+```js
+var remark = require("remark")
+var visit = require("unist-util-visit")
+
+var tree = remark().parse("Some _emphasis_, **importance**, and `code`.")
+
+visit(tree, "text", visitor)
+
+function visitor(node) {
+  console.log(node)
+}
+```
+
+Here, it finds all text nodes and will `console.log` the nodes.
+
+Now with this, we can take the AST from our plugin and get working on adding functionality for headers:
+
+```js
+const visit = require("unist-util-visit")
+
+module.exports = ({ markdownAST }, pluginOptions) => {
+  visit(markdownAST, "heading", node => {
+    // Do stuff with heading nodes
+  })
+
+  return markdownAST
+}
+```
+
+Now we will visit all heading nodes and pass it into a transformer function which we can do whatever we wish.
+
+Looking again at the AST node for header:
+
+```JSON
+{
+  "type": "heading",
+  "depth": 1,
+  "children": [
+    {
+      "type": "text",
+      "value": "Hello World!",
+      "position": {
+        "start": { "line": 1, "column": 3, "offset": 2 },
+        "end": { "line": 1, "column": 15, "offset": 14 },
+        "indent": []
+      }
+    }
+  ],
+  "position": {
+    "start": { "line": 1, "column": 1, "offset": 0 },
+    "end": { "line": 1, "column": 15, "offset": 14 },
+    "indent": []
+  }
+},
+```
+
+We have context about the text as well as what depth the header is (for instance here we have a depth of 1 which would equate to an `h1` element)
+
+with the inner function of the `visit` call, I parse out all of the text and if it will map to a h1, I set the type of the node to `html` and set the node's value to be some custom html.
```

Comment:
```suggestion
With the inner function of the `visit` call, you parse out all of the text and if it will map to a h1, you set the type of the node to `html` and set the node's value to be some custom HTML.
```